Source/NSEPSImageRep.m

```diff
@@ -26,25 +26,28 @@
    Boston, MA 02110-1301, USA.
 */ 
 
+#import "config.h"
+
 #import <Foundation/NSArray.h>
 #import <Foundation/NSCoder.h>
 #import <Foundation/NSData.h>
 #import "AppKit/NSPasteboard.h"
 #import "AppKit/NSEPSImageRep.h"
+#import "GNUstepGUI/GSImageMagickImageRep.h"
+
+@interface NSBitmapImageRep (PrivateMethods)
+- (void) _premultiply;
+@end
 
 @implementation NSEPSImageRep 
 
 + (BOOL) canInitWithData: (NSData *)data
 {
-  char buffer[2];
-
-  [data getBytes: buffer length: 2];
-
-  // Simple check for Postscript
-  if (buffer[0] == '%' && buffer[1] == '!')
-    return YES;
-  else
-    return NO;
+  NSData *header = [data subdataWithRange: NSMakeRange(0,4)];
+  NSString *str = [[NSString alloc] initWithData: header encoding: NSUTF8StringEncoding];
+  BOOL result = [str isEqualToString: @"%!PS"];
```

Comment:
If this string is correct, why don't we just get the first four bytes as above and check these?

Reply:
It seemed easier to do it this way rather than pulling out the bytes individually.  Do you feel strongly that I should make this change?

Reply:
No, I was just wondering why you switch to this slower code.

Reply:
I'm deferring this change until later.